### DISCUSSION TITLE
Accessible hover buttons

### DISCUSSION THREAD

**Discussion by appsforartists** [jsxstyle/jsxstyle#190] [Q&A] [answered]
Consider an interface with a list of images.  Each image has a delete button that removes it from the list.  Because you don't know what color the images will be, there's a scrim behind the buttons.  The scrim and the buttons appear on hover.  (The users have mice, so touch usability of hover effects is not a concern.)

You can implement this in jsxstyle like so:
```tsx
  assets.value.map(
    ({ src, width, height }) => (
      <Block
        position = 'relative'
        marginRight = { ASSET_GAP }
      >
        <img
          width = { width }
          height = { height }
          src = { src }
        />
        <Row
          position = 'absolute'
          top = { 0 }
          left = { 0 }
          bottom = { 0 }
          right = { 0 }
          background = 'linear-gradient(180deg, hsla(0 0% 0% / 12%) 0px, transparent 56px)'
          opacity = { 0 }
          hoverOpacity = { 1 }
          transition = 'opacity 150ms'
          justifyContent = 'flex-end'
          alignItems = 'flex-start'
        >
          <Block
            padding = { 12 }
            cursor = 'pointer'
            props = {
              {
                onClick() {
                  assets.value = assets.value.filter(
                    (asset) => asset.src !== src
                  )
                }
              }
            }
          >
            { deleteIcon }
          </Block>
        </Row>
      </Block>
    )
  )
}
```

This is a good first pass, but it's not keyboard accessible.  You can get closer by adding these rules to the delete `Block`:
```tsx
component = 'button'
background = 'transparent'
color = 'white'
border = 'none'
```

Now you can tab to the button, but it will remain hidden if the image isn't being hovered over.  (This is actually worse, because now you can delete the image without seeing any delete affordances.)

Is there an idiomatic way to express in jsxstyle "if my descendants have focus, change my opacity"?  I'd rather not resort of manual pointer events when CSS is supposed to handle this case.

**Comment by meyer** [comment, marked as answer]
hello there, me from the future. Sorry for neglecting this discussion; my GitHub notifications are in shambles.

This is now possible with some special syntax borrowed from Sass and, as of late, [regular old CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/Nesting_selector): the parent selector AKA "`&`". Unfortunately using an ampersand in a property name in JSX means you have to do some weird stuff with object syntax since JSX isn’t too keen on non-identifier keys.
```typescript
<Block {...{
  "&:hover .child": {
    // any valid jsxstyle styles can go here
  }
}}>
  <Block className="child">Child element</Block>
</Block>
```

I’d love computed property name support in JSX but until then, we’re stuck with this funky spread syntax thing.
